tests/Doctrine/Tests/ORM/QueryTest.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Doctrine\Tests\ORM;
+
+use Doctrine\Tests\Models\CMS\CmsUser;
+use Doctrine\Tests\OrmTestCase;
+
+/**
+ * Test case for the Query class
+ *
+ * @author Konstantin Kuklin <konstantin.kuklin@gmail.com>
+ */
+class QueryTest extends OrmTestCase
+{
+    /**
+     * @var \Doctrine\ORM\EntityManager
+     */
+    private $_em;
+
+    protected function setUp()
+    {
+        $this->_em = $this->_getTestEntityManager();
+    }
+
+    /**
+     * setResultCacheProfile should works correct with null
+     */
+    public function testSetResultCacheProfile()
+    {
+        $query = $this->_em->createQuery('SELECT u FROM ' . CmsUser::class . ' u');
+        $result = $query->setResultCacheProfile(null);
```

Comment:
This is a bit misleading, `$result` is not the query result but rather the query